AlertBox Component

U-DON opened the pull request:
NOTE: Not working; do not merge yet.

Closes department-of-veterans-affairs/vets-website#2572.

Currently unable to import the USWDS styles since Webpack can't seem to resolve the icons. I'm getting messages like this one:
```
ERROR in ./~/css-loader!./~/sass-loader!./assets/scss/rx.scss
Module not found: Error: Cannot resolve 'file' or 'directory' ../img/alerts/success.png in /home/eugene/Workspace/rx/assets/scss
 @ ./~/css-loader!./~/sass-loader!./assets/scss/rx.scss 6:41363-41399
```
Looks like the USWDS stylesheet is using relative paths to images, but Webpack isn't starting from the same path. How to get the import working?

alexose commented:
I know this isn't a great answer, but my thought would be to leave USWDS alone given that we are importing so many styles from vets.gov.  If there specific styles you need, it's probably better just to add them to a file in _partials.

My second thought is that if we do want to consume USWDS, we should consume it from a CDN.

webinista commented:
CDNJS currently hosts the USWDS' assets. The downsides as far as I can see are:

1. Incorporating too much. This would add the _entire_ USWDS library. We haven't yet removed Foundation. We have a lot of our own CSS. This would add another HTTP request (I don't _think_ we're serving over HTTP/2 yet).
1. The trustworthiness of CDNJS' servers. Since they're using CloudFlare to host, I don't think the risk is super high here.

alexose commented:
On the other hand, you get caching :)

But yes, let's err on the side of fewer CSS libraries.

U-DON commented:
So am I hearing that we should just copy-pasta only the styles we need into our own stylesheets? If the styles include image assets, do we just download directly to our `assets/img` as well?

webinista commented:
This feels like a place where it may make sense to add the styles and images to a vets-website PR. @crwallace thoughts?

crwallace commented:
In the new single repo model USWDS and all of our images will be available and part of the "va-core" styles.

U-DON commented:
Updated with alert closing functionality and [unused as of now] function to pop up alerts. Also resolved any complaints from ESLint.

Locally, I tested by including USWDS and using a placeholder alert for the initial state. Then I added some Rx-specific styles to get it to ultimately look right. But for this PR, no USWDS and no placeholder alert.

### Preview of alerts:
![screenshot from 2016-08-03 20 22 52](https://cloud.githubusercontent.com/assets/1067024/17386662/c98b3358-59ba-11e6-93b4-4fec95683da3.png)

alexose commented:
This looks great.  There might be a time where we want to move some of that functionality out of `RxRefillsApp.jsx`, but it's an easy refactor.

👍
